sympy/solvers/solveset.py

```diff
@@ -428,6 +442,54 @@ def _solve_trig(f, symbol, domain):
         return ConditionSet(symbol, Eq(f_original, 0), S.Reals)
 
 
+def _solve_trig2(f, symbol, domain):
+    """Secondary helper to solve trigonometric equations,
+    called when first helper fails """
+    from sympy import lcm, expand_trig, degree, simplify
+    f = trigsimp(f)
+    f_original = f
+    trig_functions = f.atoms(sin, cos, tan, sec, cot, csc)
+    trig_arguments = [e.args for e in trig_functions]
+    denominators = []
+    for ar in trig_arguments:
+        try:
+            poly_ar = Poly(ar, symbol)
+        except ValueError:
+            raise ValueError("give up, we can't solve if this is not a polynomial in x")
+        if poly_ar.degree() > 1:  # degree >1 still bad
+            raise ValueError("degree of variable inside polynomail should not exceed one")
+        if poly_ar.degree() == 0:  # degree 0, no x, don't care
+            continue
+        c = poly_ar.all_coeffs(x)[0]   # got the coefficient of x
+        denominators.append(Rational(c).q)
+
+    x = Dummy('x')
+    f = f.subs(symbol, 2*lcm(denominators)*x)
```

Comment:
precompute and store as `LCM_den = ilcm(*denominators)` (or similar) and use when needed below.